sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -294,5 +452,121 @@ describe(
         DisbursementScheduleStatus.Pending,
       );
     });
+
+    /**
+     * Create and save required number of notifications for student and ministry.
+     * @param notificationsCount number of notifications to create.
+     * @param student related student.
+     * @param disbursementId related disbursement id.
+     */
+    async function saveNotifications(
+      notificationsCount: number,
+      student: Student,
+      disbursementId: number,
+      daysAhead?: number,
+    ): Promise<void> {
+      while (notificationsCount > 0 && notificationsCount--) {
+        const studentNotification = createFakeNotification(
+          {
+            user: student.user,
+            auditUser: systemUsersService.systemUser,
+            notificationMessage: {
+              id: NotificationMessageType.StudentNotificationDisbursementBlocked,
+            } as NotificationMessage,
+          },
+          {
+            initialValue: {
+              messagePayload: {
+                email_address: student.user.email,
+                template_id:
+                  NotificationMessageType.StudentNotificationDisbursementBlocked,
+                personalisation: {
+                  givenNames: student.user.firstName ?? "",
+                  lastName: student.user.lastName,
+                },
+              },
+              metadata: {
+                disbursementId,
+              },
+              createdAt: daysAhead ? addDays(daysAhead) : new Date(),
+            },
+          },
+        );
+        const ministryNotification = createFakeNotification(
+          {
+            user: systemUsersService.systemUser,
+            auditUser: systemUsersService.systemUser,
+            notificationMessage: {
+              id: NotificationMessageType.MinistryNotificationDisbursementBlocked,
+            } as NotificationMessage,
+          },
+          {
+            initialValue: {
+              messagePayload: {
+                email_address: gcNotifyService.ministryToAddress(),
+                template_id:
+                  NotificationMessageType.MinistryNotificationDisbursementBlocked,
+                personalisation: {
+                  givenNames: student.user.firstName ?? "",
+                  lastName: student.user.lastName,
+                },
+              },
+              metadata: {
+                disbursementId,
+              },
+              createdAt: daysAhead ? addDays(daysAhead) : new Date(),
+            },
+          },
+        );
+        studentNotification.dateSent = new Date();
+        ministryNotification.dateSent = new Date();
+        await db.notification.save([studentNotification, ministryNotification]);
```

Comment:
The save can be done outside the loop.

Reply:
Here is the code snippet of the changes I mentioned above. 

```ts
async function saveNotifications(
      notificationsCount: number,
      student: Student,
      disbursementId: number,
      daysAhead?: number,
    ): Promise<void> {
      const notifications: Notification[] = [];
      let notificationCounter = notificationsCount;
      while (notificationCounter > 0) {
        const studentNotification = createFakeNotification(
          {
            user: student.user,
            auditUser: systemUsersService.systemUser,
            notificationMessage: {
              id: NotificationMessageType.StudentNotificationDisbursementBlocked,
            } as NotificationMessage,
          },
          {
            initialValue: {
              messagePayload: {
                email_address: student.user.email,
                template_id:
                  NotificationMessageType.StudentNotificationDisbursementBlocked,
                personalisation: {
                  givenNames: student.user.firstName ?? "",
                  lastName: student.user.lastName,
                },
              },
              metadata: {
                disbursementId,
              },
              createdAt: daysAhead ? addDays(daysAhead) : new Date(),
              dateSent: new Date(),
            },
          },
        );
        notifications.push(studentNotification);
        const ministryNotification = createFakeNotification(
          {
            user: systemUsersService.systemUser,
            auditUser: systemUsersService.systemUser,
            notificationMessage: {
              id: NotificationMessageType.MinistryNotificationDisbursementBlocked,
            } as NotificationMessage,
          },
          {
            initialValue: {
              messagePayload: {
                email_address: gcNotifyService.ministryToAddress(),
                template_id:
                  NotificationMessageType.MinistryNotificationDisbursementBlocked,
                personalisation: {
                  givenNames: student.user.firstName ?? "",
                  lastName: student.user.lastName,
                },
              },
              metadata: {
                disbursementId,
              },
              createdAt: daysAhead ? addDays(daysAhead) : new Date(),
              dateSent: new Date(),
            },
          },
        );
        notifications.push(ministryNotification);
        notificationCounter--;
      }
      await db.notification.save(notifications);
    }
```

Reply:
I was done something very similar but slightly different. Please let me know if that is acceptable.